include/unifex/type_erased_stream.hpp

```diff
@@ -104,24 +104,36 @@ struct _stream<Values...>::type final {
     virtual void start_cleanup(cleanup_receiver_base& receiver) noexcept = 0;
   };
 
+  struct next_op_base {
+    virtual ~next_op_base() = default;
+    virtual bool complete() noexcept = 0;
```

Comment:
We don't need a virtual destructor because no one's going to delete through a pointer-to-base; you can enforce that with a non-public, non-virtual, defaulted destructor.
```suggestion
    virtual bool complete() noexcept = 0;
   protected:
    next_op_base() noexcept = default;
    next_op_nase(next_op_base&&) = delete;
    ~next_op_base() = default;
```